[philbertshea] iP

philbertshea opened the pull request:
# Phil
Phil is your **personal todo list** for taking down <ins>ANY</ins> tasks you need to complete, like:
- [ ] Learning a new language (A Todo item)
- [ ] Your next assignment due two weeks later (A Deadline item)
- [ ] Joining the next talk at NUS (An Event item)

_Anyone_ can use it, including but not limited to:
1. Professionals 🥇 
2. Students 💯 
3. Retirees 👍 

View my [repo](https://github.com/philbertshea/ip) to ~~learn more~~ TRY IT NOW. 👍
- Ensure Java and Git and Gradle is installed.
- Clone the repo with `git clone https://github.com/philbertshea/ip.git`
- Run the app (GUI). `./gradlew run`

A sample code block would be:
```java
public class Deadline extends Task {
    @Override
    public String toString() {
        return "[D]" + super.toString() + " (by: "
                   + this.byDateToString(Deadline.outputFormatter) + ")";
    }
}
```

> A Goal without a plan is just a wish. - Antoine de Saint-Exupéry

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

philbertshea committed:
Refactor Duke.java to Phil.java and print required skeleton message.

philbertshea committed:
Added printOutput function and edited main in Phil.java to repeat user input until user says bye.

philbertshea committed:
Edited main to remind user every 10 inputs that they can end the conversation.

philbertshea committed:
Edit main in Phil.java to store tasks and print them using list.

philbertshea committed:
Added Task class, and edited main in Phil.java to add marking functionality.

philbertshea committed:
Edited main in Phil.java to add unmark functionality

philbertshea committed:
Edited main in Phil.java to add a one-liner describing 'Here is your list of tasks' when list is passed as input.

philbertshea committed:
Made Task abstract and added Todo, Deadline, Event subclasses to Task.

philbertshea committed:
Added TaskList class and edited Phil.java to use TaskList

philbertshea committed:
Edited Phil.java to support todo, deadline and event tasks.

philbertshea committed:
Ran text-ui-test/runtest.bat to test code, using a given input in input.txt

philbertshea committed:
Created PhilException, InvalidArgumentException and InvalidCommandException to handle errors. Updated Phil.java to handle exceptions.

philbertshea committed:
Added CommandType.DELETE_Task, edited InvalidArgumentException.java and Phil.java and TaskList.java to support removing of tasks.

philbertshea committed:
Added explanation on enum CommandType used.

philbertshea committed:
Edit Phil.java, Task.java and subclasses, TaskList.java to incorporate saving and loading

Add code before Scanner code in Phil.java to open and read data/phil.txt for saved data. Add code when bye is typed, to save tasks in TaskList to data/phil.txt.

philbertshea committed:
Edit Phil.java and Task.java to save and load task done status as well

Save one more param - status icon - into data/phil.txt per task in Task.java. Process this extra param in Phil.java.

philbertshea committed:
Merge branch 'branch-Level-7'
Merge branch-Level-7 with master to incorporate saving and loading
feature

philbertshea committed:
Edit Phil.java, Deadline.java, Event.java to process strings as DateTime if possible

Add logic within Deadline and Event constructors to parse byDate, fromDate, toDate as DateTime objects.

philbertshea committed:
Merge branch 'branch-Level-8'
Merge branch-Level-8 to master to introduce DateTime functionality

philbertshea committed:
Add Storage.java, edit Phil.java to encapsulate Storage handler

Use Storage class with load and save methods to handle loading and saving to disk.

philbertshea committed:
Add Parser.java, Ui.java and edit Phil.java to encapsulate Parsing and Ui into its separate components

Add Parser.java with processInput method that takes in a String as input, and returns the output as per previous logic. Add Ui.java with printOutput and runBot methods to handle scanning for input and printing output.

philbertshea committed:
Merge A-MoreOOP branch with master to introduce OOP functionality

Merge A-MoreOOP branch for OOP via new classes Parser, Ui and Storage

philbertshea committed:
Create phil package, move all java files to phil package

philbertshea committed:
Merge branch-A-Packages to master

philbertshea committed:
Merge add-gradle-support to master

philbertshea committed:
Added TodoTest.java and ParserTest.java with testing methods for JUnit Tests

philbertshea committed:
Merge branch-A-JUnit with master for JUnit tests

philbertshea committed:
Edit settings in build.gradle, build jar using ShadowJar and try running phil.jar using java -jar method

philbertshea committed:
Merge branch-A-Jar with master

philbertshea committed:
Add Javadoc comments for all or most public classes and fields

Also made edits to access modifiers of classes to ensure OOP information hiding

philbertshea committed:
Edit Deadline, Event, InvalidArgumentException, Parser files to avoid excessively long lines.

philbertshea committed:
Edit Parser, Task and TaskList files to add finding logic

Add descContains method in Task.java. Add filteredTasksToString method in TaskList.java. Add logic when input starts with find in Parser.java.

philbertshea committed:
Merge Branch-A-JavaDoc with master

philbertshea committed:
Merge branch-A-CodingStandard with master

philbertshea committed:
Merge branch-Level-9 with master

philbertshea committed:
Update JavaDoc based on coding standards

philbertshea committed:
Refactor Phil.java to PhilTextUI.java, add multiple classes to incorporate GUI Functionality

To incorporate GUI Functionality, add DialogBox.java, Launcher.java, MainWindow.java, PhilMain.java, and refactor Phil.java.\n Add resources/images folder with two images, and resources/view with two fxml files.

philbertshea committed:
Merge branch 'branch-Level-10'
Merge branch-Level-10 into master to incorporate GUI Functionality

philbertshea committed:
Update build.gradle and add config/checkstyle files to support checkstyle checks

philbertshea committed:
Make styling changes to multiple files to pass all CheckStyle tests

philbertshea committed:
Merge branch-A-CheckStyle with master to satisfy CheckStyle requirements

philbertshea committed:
Edit DialogBox and MainWindow to add flipping of dialog box logic

philbertshea committed:
MainWindow.java: Remove unused imports

Unused imports cause the CheckStyle test to fail.
This violates the coding standard that all imports should be used at
least once.

Let's
* remove the unused imports to satisfy the CheckStyle test

philbertshea committed:
DialogBox.java: Add pictureHeight to constructor

User creating a DialogBox might want to edit the Size of the image.

Let's
* Add parameter pictureHeight to DialogBox constructor
* Use pictureHeight as parameter to set FitWidth and FitHeight of
  displayPicture
* Update use of DialogBox constructor in MainWindow.java

philbertshea committed:
DialogBox.fxml: Add padding to Label

Text looks too close to image of user or chatbot,
which is not optimal for reading and delivers a poorer user experience.

Let's
* add a padding of 10 (left) and 10 (right) to the Label

philbertshea committed:
MainWindow.fxml: Add background color

Chatbot interface looks dull with the default colour.

Let's
* add green background colour #CAE0BC to the VBox

philbertshea committed:
DialogBox.fxml: Add CSS styling to Label

Text in DialogBox doesn't look clear.

Let's
* add background colour and border to Label in DialogBox

philbertshea committed:
DialogBox.java: Add font-size CSS logic

Font size should change based on length of text in dialog.

Let's
* Remove font size from style block in DialogBox.fxml
* Add logic to adjust font size based on length of text
in DialogBox.java

philbertshea committed:
README.md: Edit README content

README content is outdated, based on the initial Duke template.

Let's
* update the content to be more specific to the Phil bot

philbertshea committed:
Repackage classes into subpackages

Classes were all packed into one phil class.
It is less easy to keep classes organised especially if the app grows.

Let's
* Create five new subpackages under phil - exception, main, model,
  storage, ui
* Repackage existing classes into these packages
* Refactor the files and build.gradle to accommodate this change

jonzyyyy commented on src/main/java/phil/main/Parser.java:
```diff
@@ -0,0 +1,122 @@
+                String fromDate = String.join(" ", inputArgs.subList(fromIndex + 1, toIndex));
+                String toDate = String.join(" ", inputArgs.subList(toIndex + 1, inputArgs.size()));
+                return this.taskList.addTask(new Event(description, fromDate, toDate));
+
```
Perhaps you forgot to delete the empty space here?

philbertshea replied:
Looks like it. Thanks for the reminder.

jonzyyyy commented on src/main/java/phil/main/Phil.java:
```diff
@@ -0,0 +1,46 @@
+
+/**
+ * Represents the Phil instance that stores the parser
+ * and a function to get the response from parser based on user input.
```
Noticed some inconsistencies in your class javadoc comments - Some have a return line after and some don't. Would be better to keep a consistent formatting

(comparing to the Parser class)

philbertshea replied:
Got it, I've checked and updated the Javadoc comments to largely follow a similar formatting. Thanks.

jonzyyyy commented on src/main/java/phil/model/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+     * If a LocalDateTime was provided as the bydate, it formats it based on the formatter provided.
+     * Else, the String bydate is returned.
+     *
+     * @param formatter formattr to format the LocalDateTime object representing the by date to.
```
Small spelling error

philbertshea replied:
Yup, something I somehow missed out. Corrected it. Thanks

jonzyyyy commented on src/main/java/phil/storage/Storage.java:
```diff
@@ -0,0 +1,87 @@
+package phil.storage;
+
+import phil.model.*;
```
Perhaps leave out the asteriks in this course? haha

philbertshea replied:
Yup, thanks for spotting this.

jonzyyyy reviewed (commented):
Great work!! Only minor corrections.

philbertshea committed:
Edit multiple files after reviewer's comments

From reviewer jonzyyyy's review, there are multiple
coding standard violations, particularly pertaining to
inconsistency in JavaDoc comments.

I also discovered additional inconsistencies in JavaDoc comments.

Let's
* Address the coding standard violations pointed out by the reviewer
* Update the JavaDoc comments to make the formatting consistent

philbertshea committed:
Add assertions in Deadline.java, Event.java

Deadline.java and Event.java functionality depend on
byDate OR byDateInDateTime being non-null. There is no
way to guarantee that without an assertion, leading
to potential errors.

Let's
* Add assertion in Deadline.java
* Add assertion in Event.java

philbertshea committed:
Add assertions in Storage, Parser and Phil

Parser.java and Phil.java work on the
critical assumptions that parser, storage attributes are not null.
Storage.java assumes the directory and file at dirPath and dataPath
exists.
There are no checks to ensure these are true.

Let's
* Add assertions in Storage.java to ensure dirPath and dataPath exist
* Add assertions in Parser.java and Phil.java to ensure parser, storage
  exist

philbertshea committed:
Edit static variables naming to correct case

Static variables should be in SCREAMING_SNAKE_CASE
instead of camelCase.

Let's
* Rename static variables in Deadline.java and Event.java

philbertshea committed:
Merge pull request #1 from philbertshea/branch-A-Assertions

Merge branch-A-Assertions to master

philbertshea committed:
Merge branch 'master' into branch-A-CodeQuality
Changes to master from merged PR #1 initiated by
branch-A-Assertions are not reflected.

Let's
* Merge master branch into branch-A-CodeQuality

philbertshea committed:
Remove unused exception in MainWindow.java

MainWindow's handleUserInput throws an exception
but no part of the code throws that exception.

Let's
* Remove the unused exception in handleUserInput

philbertshea committed:
MainWindow.java: Edit naming of method

handleUserInput is a vague naming.
The method actually adds the input and output dialogs
upon the user giving an input.

Let's
* Rename the method to addInputAndOutputDialogs

philbertshea committed:
Parser.java: Abstract repeated logic

There are repeated code segments in Parser.java.

Let's
* Abstract these segments into separate methods

philbertshea committed:
Update javadoc comments in Parser.java

Javadoc comments are not updated for the new methods.

Let's
* Add or edit comments to adhere to JavaDoc standards

philbertshea committed:
Merge pull request #2 from philbertshea/branch-A-CodeQuality

Make changes for Code Quality improvements.

philbertshea committed:
Add Note.java, NoteList.java, modify others

Some items may be notes, may not be tasks.
These are simply stuff we want to take down for reference.
Our bot cannot accommodate for these items.

In particular, these items do not need to be marked as done
or undone. They have no related dates or times.
They are just textual information.

Let's
* Add Note.java to represent a Note instance
* Add NoteList.java to represent a list of notes
* Add logic in Parser to add a note, delete a note,
and list all notes
* Update Storage.java, Phil.java, PhilTextUI.java to account
for additional storage of notes in a separate file
* Add corresponding enums in CommandType
* Update IllegalCommandException and IllegalArgumentException
for the possible exceptions from the related actions
* Add JavaDoc comments for the new classes and methods

philbertshea committed:
Merge branch 'branch-D-Notes'
Merge branch for note taking functionality into master.

philbertshea committed:
Update incorrect info on InvalidCommandException

InvalidCommandException gives a list of commands that are accepted
together with their use cases. Discovered one command was incorrectly
written as view-note instead of list-note.

Let's
* Correct the instructions to list-note

philbertshea committed:
Add Command and subclasses for OOP

Commands are currently not abstracted out.
This leads to an excessively long generateStringOutputFromInput method
within the Parser.java file.

Let's
* Add TaskCommand and NoteCommand abstract classes.
* Add subclasses to these two abstract classes.
* Edit Parser.java to update the logic of
  generateStringOutputFromInput.

philbertshea committed:
Add personalised image of chatbot

Chatbot image looks generic.

Let's
* Add a new image Phil.png into resources/images
* Update philImage in MainWindow.java to get from the new resource

philbertshea committed:
Add Ui.png screenshot to docs

Users may want to know how the app looks like, before trying the app.

Let's
* Take a screenshot of the app
* Add it into docs folder as Ui.png

philbertshea committed:
README.md: Add description for todo, event tasks

README.md does not have sufficient info about creating tasks.

Let's
* Add description for todo and event tasks

philbertshea committed:
Event.java, Deadline.java: Edit input formatter

Users may not want to type out the full date in MM/dd/yyyy format.

Let's
* Shorten the accepted format to "M/d/yy"

philbertshea committed:
README.md: Add descriptions of remaining commands

README.md is still inadequate as it does not explain
other commands like creating deadline, deleting and
marking.

Let's
* Add the documentation for the other commands

philbertshea committed:
Add multiple JUnit Test classes

Only two classes have JUnit tests.
The number of tests is inadequate.

Let's
* Refactor existing tests into main and model packages
* Update tests in TodoTest.java
* Add tests for Deadline, Event, Note, NoteList, TaskList

philbertshea committed:
Merge branch 'branch-A-MoreTesting'
Merge branch-A-MoreTesting into master

philbertshea committed:
Add a reminder when tasklist/notelist gets big

Phil doesn't like to procrastinate.
We should warn the user when he has more than 10 tasks or notes.

Let's
* Edit NoteList and TaskList's toString to add a line
warning the user of "too many tasks/notes" when the
number of tasks/notes exceeds 10

philbertshea committed:
TaskList.java: Update filteredTasksToString

Users may want to use the find function to get the
real index of the corresponding tasks with a search term
so they can mark it as done or undone.
The current filteredTasksToString uses a temporary count
as the index displayed from find command instead.

Let's
* Edit filteredTasksToString to show the actual
index of tasks matching the search term

philbertshea committed:
Merge branch 'branch-A-Personality'
Merge branch-A-Personality into master

philbertshea committed:
Update phil.txt

Phil.txt seems to be corrupted.

Let's
* Update phil.txt

philbertshea committed:
Edit multiple files to follow CheckStyle

CheckStyle tests failed, which indicate some sort
of coding standard violation.

Let's
* Run CheckStyle tests
* Identify and rectify the problematic code
violating the CheckStyle standards

philbertshea committed:
Edit app icon to be images/Phil.png

Default app icon is not relevant to the Phil chatbot.

Let's
* Edit PhilMain.java to add the resource /images/Phil.png
as the stage icon

philbertshea committed:
InvalidCommandException.java: Update message

Message shown when exception is returned is too long,
primarily due to the redundant new lines.

Let's
* Update the message to remove the new lines when
an InvalidCommandException is thrown

philbertshea committed:
Deadline.java: Remove message printed

A message used for debugging is being printed.
This message is no longer needed.

Let's
* Update Deadline.java to remove the printing
of the message whenever the bydate is not parsable

philbertshea committed:
MainWindow.fxml: Anchor to allow resizing

Window would not resize properly, and default size
is too small for normal use.

Let's
* Add AnchorPane anchors to allow proper resizing of the window

philbertshea committed:
Update Ui.png to reflect new interface

Current Ui.png does not reflect changes in interface,
such as the larger size on opening.

Let's
* Reupload the Ui.png image

philbertshea committed:
Update JavaDoc comments

Some JavaDoc comments did not follow the
imperative mood as specified by the
coding standards.

Let's
* Update these comments to follow the coding standards

philbertshea committed:
Update InvalidCommandException message

InvalidCommandException message has some errors.
The suggested format of deadline and event do not work.

Let's
* Update the message of the InvalidCommandException
to show the correct commands
